<a href="https://colab.research.google.com/github/Swethaa-sri/matplotlib/blob/main/euclidean__vs_manhattan_D15.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [46]:
import numpy as np
import pandas as pd

In [47]:
data_train = pd.read_csv('glass_identi_train.csv')
data_test = pd.read_csv('glass_identi_test.csv')

In [48]:
data_train.head()

Id number       RI     Na    Mg    Al  ...     K    Ca   Ba   Fe  Type of glass
0          1  1.52101  13.64  4.49  1.10  ...  0.06  8.75  0.0  0.0              1
1          2  1.51761  13.89  3.60  1.36  ...  0.48  7.83  0.0  0.0              1
2          3  1.51618  13.53  3.55  1.54  ...  0.39  7.78  0.0  0.0              1
3          4  1.51766  13.21  3.69  1.29  ...  0.57  8.22  0.0  0.0              1
4          5  1.51742  13.27  3.62  1.24  ...  0.55  8.07  0.0  0.0              1

[5 rows x 11 columns]

In [49]:
data_test.head()

Id number       RI     Na    Mg    Al  ...     K     Ca   Ba    Fe  Type of glass
0         68  1.52152  13.05  3.65  0.87  ...  0.19   9.85  0.0  0.17              1
1         69  1.52152  13.12  3.58  0.90  ...  0.23   9.82  0.0  0.16              1
2         70  1.52300  13.31  3.58  0.82  ...  0.12  10.17  0.0  0.03              1
3        144  1.51709  13.00  3.47  1.79  ...  0.66   8.18  0.0  0.00              2
4        145  1.51660  12.99  3.18  1.23  ...  0.58   8.81  0.0  0.24              2

[5 rows x 11 columns]

In [50]:
data_train.describe()

Id number          RI  ...          Fe  Type of glass
count  196.000000  196.000000  ...  196.000000     196.000000
mean   102.862245    1.518295  ...    0.050255       2.668367
std     61.195663    0.003055  ...    0.086359       2.062416
min      1.000000    1.511310  ...    0.000000       1.000000
25%     49.750000    1.516458  ...    0.000000       1.000000
50%    101.500000    1.517630  ...    0.000000       2.000000
75%    153.250000    1.518985  ...    0.090000       3.000000
max    211.000000    1.533930  ...    0.340000       7.000000

[8 rows x 11 columns]

In [51]:
data_train.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 196 entries, 0 to 195
Data columns (total 11 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Id number      196 non-null    int64  
 1   RI             196 non-null    float64
 2   Na             196 non-null    float64
 3   Mg             196 non-null    float64
 4   Al             196 non-null    float64
 5   Si             196 non-null    float64
 6   K              196 non-null    float64
 7   Ca             196 non-null    float64
 8   Ba             196 non-null    float64
 9   Fe             196 non-null    float64
 10  Type of glass  196 non-null    int64  
dtypes: float64(9), int64(2)
memory usage: 17.0 KB


In [63]:
##  Dropping irrelevant fields for predicted values
data_train = data_train.drop(['Id number'], axis = 1)
data_test = data_test.drop(['Id number'], axis = 1)

In [64]:
## X_train , y_train 
X_train = data_train.iloc[:,:-1].values
y_train = data_train.iloc[:,-1].values

In [65]:
## X_test, y_test
X_test = data_test.iloc[:,:-1].values
y_test = data_test.iloc[:,-1].values

In [66]:
## Standardisation
from sklearn.preprocessing import StandardScaler
sc = StandardScaler()
X_train = sc.fit_transform(X_train)
X_test = sc.transform(X_test)

In [67]:
# KNN model using Square Euclidean
from sklearn.neighbors import KNeighborsClassifier
classifier = KNeighborsClassifier(n_neighbors = 8, metric='euclidean')
classifier.fit(X_train, y_train)

KNeighborsClassifier(metric='euclidean', n_neighbors=8)

In [68]:
## prediction
y_pred = classifier.predict(X_test)
#print(np.concatenate((y_pred.reshape(len(y_pred),1), y_test.reshape(len(y_test),1)),1))

In [69]:
## classification report
from sklearn.metrics import accuracy_score,classification_report
#print(accuracy_score(y_test, y_pred))
print('Classification Report')
print(classification_report(y_test, y_pred))

Classification Report
              precision    recall  f1-score   support

           1       0.75      1.00      0.86         3
           2       0.29      0.67      0.40         3
           3       0.00      0.00      0.00         3
           5       1.00      0.33      0.50         3
           6       1.00      0.67      0.80         3
           7       0.75      1.00      0.86         3

    accuracy                           0.61        18
   macro avg       0.63      0.61      0.57        18
weighted avg       0.63      0.61      0.57        18



/usr/local/lib/python3.7/dist-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
/usr/local/lib/python3.7/dist-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
/usr/local/lib/python3.7/dist-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))


In [70]:
# KNN model using manhattan
from sklearn.neighbors import KNeighborsClassifier
classifier = KNeighborsClassifier(n_neighbors = 8, metric='manhattan')
classifier.fit(X_train, y_train)

KNeighborsClassifier(metric='manhattan', n_neighbors=8)

In [71]:
# prediction
y_pred = classifier.predict(X_test)

In [72]:
# classification report
from sklearn.metrics import accuracy_score,classification_report
accuracy_score(y_test, y_pred)
print('Classification Report')
print(classification_report(y_test, y_pred))

Classification Report
              precision    recall  f1-score   support

           1       0.43      1.00      0.60         3
           2       0.33      0.67      0.44         3
           3       0.00      0.00      0.00         3
           5       0.00      0.00      0.00         3
           6       1.00      0.33      0.50         3
           7       0.75      1.00      0.86         3

    accuracy                           0.50        18
   macro avg       0.42      0.50      0.40        18
weighted avg       0.42      0.50      0.40        18



/usr/local/lib/python3.7/dist-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
/usr/local/lib/python3.7/dist-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
/usr/local/lib/python3.7/dist-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))


##### The KNN model with euclidean as distance metric gives an accuracy of 61% while with manhattan as distance metric gives an accuracy of 50%.
##### Thus the performance of KNN model with euclidean distance metice is better than the performance of the model with manhattan distance metric. 